hangman.py

```diff
@@ -0,0 +1,137 @@
+import random
+import os
+import pickle
+
+### FUNCTIONS ###
+
+
+def display_title():
+    os.system('cls')
+    print("################################")
+    print("#### You're playing hangman ####")
+    print("################################")
+
+
+def get_choice():
+    print("\nEnter [1] to play Hangman. ")
+    print("Enter [2] to add your word to the pool.")
+    print("Enter [3] to see pool of words.")
+    print("Enter [q] to quit game.")
+    choice = input("What would you like to do? ")
+    return choice
+
+
+def load_words():
+    # This function loads names from a file, and puts them in the list 'names'.
+    #  If the file doesn't exist, it creates an empty list.
+    try:
+        file_object = open('words.txt', 'rb')
+        names = pickle.load(file_object)
+        file_object.close()
+        return names
+    except Exception as e:
+        print(e)
+        return []
+
+
+def quit():
+    # This function dumps the names into a file, and prints a quit message.
+    try:
+        file_object = open('words.txt', 'wb')
+        pickle.dump(words, file_object)
+        file_object.close()
+        print("\nThanks for playing. I will remember these words.")
+    except Exception as e:
+        print("\nThanks for playing. I won't be able to remember these words.")
+        print(e)
+
+
+def hangman():
+    # Initialize hangman
+    # I know, this part is redundant, but I like it.
+    guess_word = random.choice(words)
+    print("\nCan you guess?")
+    number_of_underscores = len(guess_word)
+    print(number_of_underscores * " __")
+    print(f"Number of letters: {len(guess_word)}")
+
+    count_wrong_guess = 0  # counts number of wrong guesses
+    bad_letters = []  # list of wrong letters
+    good_letters = []  # list of good letters
+    secret_word = "__"  # final result that the player sees
+    game_round = 0  # number of played rounds, total: 8
+
+    while (count_wrong_guess < 8) and ("__" in secret_word):
+        # user chooses a letter
+        guess_letter = input("\nGuess a letter: ")
```

Comment:
You use `guess_word` for the word that user has to guess, and here you use `guess_letter` for the letter user tried to guess. That is misdirecting, as one would think that `guess_letter` is the letter from `guess_word` or smth like that, due to same prefix, but really they are in some way opposite to teach other -> one is being guessed, one is the guess. So we should use more differentiating names here.